zebra-rpc/src/lib.rs

```diff
@@ -15,3 +15,82 @@ pub mod indexer;
+// which is split mostly to keep file sizes small. Additionally, we re-export
+// types from other crates which are exposed in the API.
+//
+// TODO:
+// - Consider exporting functions here as well
+// - Expand documentation?
+// - Move `hex_data` and `trees` modules to/under the `types` module and qualify type refs in `methods`?
+pub mod client {
+    /// Types needed by clients of Zebra's RPC server
+    pub mod types {
+        pub use zebra_chain::{
+            sapling::NotSmallOrderValueCommitment,
+            serialization::{ZcashDeserialize, ZcashSerialize},
+            subtree::NoteCommitmentSubtreeIndex,
+            transparent::Script,
+            work::difficulty::{CompactDifficulty, ExpandedDifficulty},
+        };
+        pub use zebra_state::OutputIndex;
+
+        pub use crate::methods::{
+            hex_data::HexData,
+            trees::{
+                Commitments, GetSubtreesByIndexResponse, GetTreestateResponse, SubtreeRpcData,
+                Treestate,
+            },
+            types::{
+                default_roots::DefaultRoots,
+                get_block_template::{
+                    GetBlockTemplateCapability, GetBlockTemplateParameters,
+                    GetBlockTemplateRequestMode, GetBlockTemplateResponse, ProposalResponse,
+                    TemplateResponse, TimeSource,
+                },
+                get_mining_info::Response as GetMiningInfoResponse,
+                get_raw_mempool::GetRawMempoolResponse,
+                get_raw_mempool::MempoolObject,
+                peer_info::{GetPeerInfoResponse, PeerInfo},
+                submit_block::{SubmitBlockErrorResponse, SubmitBlockResponse},
+                subsidy::{BlockSubsidy, FundingStream, GetBlockSubsidyResponse},
+                transaction::{
+                    Input, Orchard, OrchardAction, Output, ScriptPubKey, ScriptSig, ShieldedOutput,
+                    ShieldedSpend, TransactionObject, TransactionTemplate,
+                },
+                unified_address::ZListUnifiedReceiversResponse,
+                validate_address::ValidateAddressResponse,
+                z_validate_address::{ZValidateAddressResponse, ZValidateAddressType},
+            },
+            BlockHeaderObject, BlockObject, GetAddressBalanceRequest, GetAddressBalanceResponse,
+            GetAddressTxIdsRequest, GetAddressUtxosResponse, GetBlockChainInfoResponse,
+            GetBlockHashResponse, GetBlockHeaderResponse, GetBlockHeightAndHashResponse,
+            GetBlockResponse, GetBlockTransaction, GetBlockTrees, GetInfoResponse,
+            GetRawTransactionResponse, Hash, SendRawTransactionResponse, Utxo,
+        };
+
+        // TODO: Move other deprecated type names here.
+        #[deprecated(note = "Use `GetSubtreesByIndexResponse` instead.")]
+        pub use GetSubtreesByIndexResponse as GetSubtrees;
+
+        #[deprecated(note = "Use `GetTreestateResponse` instead.")]
+        pub use GetTreestateResponse as GetTreeState;
```

Comment:
To be valuable the deprecated types should keep their original import paths, but that should be doable, but if we were to do this then it would make sense to create deprecated types for _all_ types since they are all moving paths... and that is probably too much work, so it seems easier to just break everything and not create these deprecated types names.